hw/scsi/scsi-disk.c

```diff
@@ -570,12 +573,68 @@ static uint8_t *scsi_get_buf(SCSIRequest *req)
     return (uint8_t *)r->iov.iov_base;
 }
 
+typedef struct __page
+{
+    uint8_t page_code;
+    uint8_t sub_page_code;
+    uint16_t offset;
+    uint16_t length;
+}PageIndex;
+
+typedef struct __pages
+{
+    uint16_t number;
+    PageIndex pi;
+}PagesHeader;
+
+typedef struct __file
+{
+    uint16_t inquiry_offset;
+    uint16_t mode_offset;
```

Comment:
Yes, page type == 0 is inquiry and 1 is mode page.
The file layout is as below:
| file header | inquiry page header | inq page1 | inq page2| ...| standard inq page | mode page header | mode page1 | mode page2 | ...| checksum | 

In the page header, the structure is:
| number of pages |  page index1 | page index2 | ... |